### PR TITLE
Oxidation fix unknown status deserialization

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs [hunk 1]
```diff
@@ -146,6 +152,21 @@ fn quote_reps(_reps: &[Rep], types: &HashMap<String, String>) -> TokenStream {
     reps
 }
 
+fn quote_valid_statutes(_reps: &[Rep]) -> TokenStream {
```
Comment (touilleMan) [inline oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:155]: "quote_valid_statuses" ?

File: oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs [hunk 2]
```diff
@@ -311,17 +333,24 @@ impl Cmds {
...
-                                let data = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
-                                Self::UnknownStatus {
-                                    _status: data.status,
-                                    reason: data.reason,
+                                let unknown_status = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
+
+                                match unknown_status.status.as_str() {
+                                    "ok" => data,
+                                    #valid_statutes
+                                    _ => Ok(Self::UnknownStatus {
```
Comment (touilleMan) [inline oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:348]: why not `Ok(unknwon_status)` ?

Comment (TimeEngineer) [reply oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:348]: What do you mean ?

Comment (touilleMan) [reply oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:348]: replace `Ok(...)` body by my stuff

File: oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs [hunk 3]
```diff
@@ -311,17 +333,24 @@ impl Cmds {
...
                                 // Instead we have this additional deserialization attempt fallback
-                                let data = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
-                                Self::UnknownStatus {
-                                    _status: data.status,
-                                    reason: data.reason,
+                                let unknown_status = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
+
+                                match unknown_status.status.as_str() {
+                                    "ok" => data,
+                                    #valid_statuses
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:347]: I have changed

File: oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs (hunk 2 above)
Comment (TimeEngineer) [reply oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs:347]: Not the same thing, `Self::UnknownStatus` is part of the `Enum`, `unknown_status` is not

**Review by touilleMan — APPROVED** [review]

**Pull request merged**